### ISSUE TITLE
Email fails regex validation

### ISSUE THREAD

**Issue by yulric** [aws/amazon-cognito-identity-js#446]
Hi there,
I'm using a CustomMessage trigger to send an html email but I aways get the error that the emails fails to meet the regex [\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*\\{####\\}[\p{L}\p{M}\p{S}\p{N}\p{P}\s*] constraint even though I can see the {####} in the email.

I can post the email here if needed

**Comment by vlekakis** [comment]
Hello, 

Yes it will be helpful if we had the email, and I can take a look. 
Let me know if you cannot post it here to find another solution

**Comment by yulric** [comment]
[cognito_email.txt](https://github.com/aws/amazon-cognito-identity-js/files/1075986/cognito_email.txt)

Here it is. If you search for {####} you can see that it's in there

**Comment by vlekakis** [comment]
Can you provide me with a request-id, AWS region and a timestamp. I want to see something more in our logs

**Comment by yulric** [comment]
RequestId: 859a0582-51df-11e7-acf4-b75e51d1bc0e
Timestamp: 2017-06-15T15:30:12.322Z
AWS Region: us-east-1, North Virginia

Thanks

**Comment by vlekakis** [comment]
The message needs to satisfy this regex:
[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*\{####\}[\p{L}\p{M}\p{S}\p{N}\p{P}\s\*]\*

**Comment by yulric** [comment]
Please look at this link https://regex101.com/r/e3mUJY/2

When I put the email I sent you along with the above regex it says it found 3 matches

**Comment by vlekakis** [comment]
Seems that the email body that reaches the service does not pass the regex
Are passing the same thing?

**Comment by vlekakis** [comment]
Also we need to make sure that the email contains everything that is URL safe.
In other words are the special characters you pass there escaped properly?

**Comment by yulric** [comment]
The file that I attached has the html string which is consoled from the Lambda function
Is there a regex that I can use to check for those special characters?

**Comment by vlekakis** [comment]
I'm sorry I'm not sure I was clear: Have you URL-encode the message that you sent to the service?

**Comment by yulric** [comment]
No I have not, I will give that a try and get back to you. Thanks

**Comment by yulric** [comment]
URL encoding works but when I get the email it has the url-encoded string in it i.e. aws isn't url decoding the string and sending it in the email

I attached the url encoded email I'm setting to the emailMessage variable in the lambda function

[cognito_email_url_encoded.txt](https://github.com/aws/amazon-cognito-identity-js/files/1081606/cognito_email_url_encoded.txt)

**Comment by vlekakis** [comment]
Yes this is true. We will take this as a feature for future releases.

**Comment by yulric** [comment]
Is there any solution in the meantime?

**Comment by vlekakis** [comment]
Modifying your message, i would think

**Comment by yulric** [comment]
I'm not sure what modification needs to be done though, is there list of special characters somewhere I can look up?

**Comment by k3karthic** [comment]
Ran into this issue recently while trying to use custom HTML in the CustomMessage trigger. For me, the problem turned out to be Unicode [U+00A0](http://www.fileformat.info/info/unicode/char/00a0/index.htm) (No-Break Space). 

I did not need to URL encode the HTML. Stripping this character from the html resolved the error.
`html = html.replace(/\u00A0/g, '');`

I used the [MJML](https://mjml.io/try-it-live) demo page to generate my email HTML, so am not sure why this particular character is generated for some templates and not others.

Hopefully, this helps others googling this issue.
